### PR TITLE
added common mistakes list for topic equals and hashcode

### PR DESCRIPTION
**Pull request by RomanDubovskyi**

### CODE REVIEW

File: java-core/equals-and-hashcode/clone.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+### Common mistakes
+
+#### Ignoring encapsulation
+Remember to use access modifiers with all class elements. We should explicitly control the access level to 
+different parts of our code.
+
+#### Creating a shallow copy of object's field
+If you copy an object of class `Car` with field `Engine` inside - keep in mind that `engine` may have non-primitive fields
+on its own. So how will they be copied?  What influence does it make on our  `car` copy? Is it deep or shallow?
```
Comment (boroda4436) [inline java-core/equals-and-hashcode/clone.md:9]:
```suggestion
on its own. So how will they be copied?  What influence does it make on our `car` copy? Is it deep or shallow?
```

**Review by boroda4436 — APPROVED** [review]

File: java-core/equals-and-hashcode/clone.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+// Other methods
+
+private int getInitialSalary(Person employee); {...}
+```
+``` 
+Refactored code: 
+
+//Contstructor and other class elements..
```
Comment (KseniiaMakarova) [inline java-core/equals-and-hashcode/clone.md:38]:
```suggestion
//Constructor and other class elements..
```

File: java-core/equals-and-hashcode/clone.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+a common task they should be located near to each other independently of their accessibility.  
+Let's consider the following situation: we need to calculate the salary of an employee. In our class we use
+3 methods to achieve desired result: `getMonthlySalary()`, `getInitialSalary()` and `getMonthlyBonus()` methods.  
+How can we order our methods?   
+``` 
+Bad example: 
+
+private int getMonthlyBonus(Person employe); {...}
+
+//Contstructor and some methods..
```
Comment (KseniiaMakarova) [inline java-core/equals-and-hashcode/clone.md:23]:
```suggestion
//Constructor and some methods..
```

File: java-core/equals-and-hashcode/clone.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+public int getMonthlySalary(Person employee) {
+    int initialSalary = getInitiallySalary(employee);
```
Comment (KseniiaMakarova) [inline java-core/equals-and-hashcode/clone.md:26]:
```suggestion
    int initialSalary = getInitialSalary(employee);
```

File: java-core/equals-and-hashcode/clone.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+private int getInitialSalary(Person employee); {...}
+```
+``` 
+Refactored code: 
+
+//Contstructor and other class elements..
+
+public int getMonthlySalary(Person employee) {
+    int initialSalary = getInitiallySalary(employee);
```
Comment (KseniiaMakarova) [inline java-core/equals-and-hashcode/clone.md:41]:
```suggestion
    int initialSalary = getInitialSalary(employee);
```

File: java-core/equals-and-hashcode/clone.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+Your class should have a clear readable structure and order of elements. Generally, order
+class methods by functionality, meaning if you have several methods that are involved in completing 
+a common task they should be located near to each other independently of their accessibility.  
+Let's consider the following situation: we need to calculate the salary of an employee. In our class we use
+3 methods to achieve desired result: `getMonthlySalary()`, `getInitialSalary()` and `getMonthlyBonus()` methods.  
+How can we order our methods?   
+``` 
+Bad example: 
+
+private int getMonthlyBonus(Person employe); {...}
```
Comment (KseniiaMakarova) [inline java-core/equals-and-hashcode/clone.md:21]:
```suggestion
private int getMonthlyBonus(Person employee) {...}
```
same for other methods in the examples

File: java-core/equals-and-hashcode/equals-and-hashcode.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+ return a == b;
+```
+
+#### Casting input object every time you compare each set of fields
+In method `equals` let's not cast input object always during field comparison but instead, you can
+create a new object and cast it to the needed type just once.
+
+#### Comparing types with `instanceof` or `getClass()`
+You should choose which of these options will suit us the best. What's the difference between them
+anyway. Try to figure it out on your own, but if you stuck, here's a [hint](https://overcoder.net/q/59643/instanceof-vs-getclass#1971415)
```
Comment (KseniiaMakarova) [inline java-core/equals-and-hashcode/equals-and-hashcode.md:22]:
```suggestion
anyway. Try to figure it out on your own, but if you're stuck, here's a [hint](https://overcoder.net/q/59643/instanceof-vs-getclass#1971415)
```

File: java-core/equals-and-hashcode/factorial.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        return 0;
+    } else {
+        return 1;
+    }
+}
+```
+```
+Refactored code: 
+ public int returnModulus(int input) {
+    if (intpu % 2 == 0) {
```
Comment (KseniiaMakarova) [inline java-core/equals-and-hashcode/factorial.md:18]:
```suggestion
    if (input % 2 == 0) {
```

File: java-core/equals-and-hashcode/fibonacci.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        return 0;
+    } else {
+        return 1;
+    }
+}
+```
+```
+Refactored code: 
+ public int returnModulus(int input) {
+    if (intpu % 2 == 0) {
```
Comment (KseniiaMakarova) [inline java-core/equals-and-hashcode/fibonacci.md:18]:
```suggestion
    if (input % 2 == 0) {
```

**Review by Sofasmile — APPROVED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**